pkg/reduce/reduce.go

```diff
@@ -148,8 +152,16 @@ func (d *DataForward) forwardAChunk(ctx context.Context) {
 	// fetch watermark using the first element's watermark, because we assign the watermark to all other
 	// elements in the batch based on the watermark we fetch from 0th offset.
 	processorWM := d.watermarkFetcher.GetWatermark(readMessages[0].ReadOffset)
-	for _, m := range readMessages {
-		m.Watermark = time.Time(processorWM)
+	if d.keyed {
```

Comment:
we could push the if/else inside of for loop.

Reply:
done